modules/account-lib/src/coin/sol/utils.ts

```diff
@@ -0,0 +1,93 @@
+import { Keypair, PublicKey } from '@solana/web3.js';
+import bs58 from 'bs58';
+import BigNumber from 'bignumber.js';
+import { BaseUtils } from '../baseCoin';
+
+const ENCODED_BLOCK_HASH_LENGTH = 44;
+const DECODED_BLOCK_HASH_LENGTH = 32;
+const ENCODED_ADDRESS_LENGTH = 44;
+const DECODED_ADDRESS_LENGTH = 32;
+const DECODED_SIGNATURE_LENGTH = 64;
+export class Utils implements BaseUtils {
+  /** @inheritdoc */
+  isValidAddress(address: string): boolean {
+    try {
+      return (
+        !!address &&
+        new RegExp('[1-9A-HJ-NP-Za-km-z]').test(address) &&
+        address.length === ENCODED_ADDRESS_LENGTH &&
+        bs58.decode(address).length === DECODED_ADDRESS_LENGTH
+      );
+    } catch (e) {
+      return false;
+    }
+  }
+
+  /** @inheritdoc */
+  isValidBlockId(hash: string): boolean {
+    return (
+      !!hash &&
+      new RegExp('[1-9A-HJ-NP-Za-km-z]').test(hash) &&
+      hash.length === ENCODED_BLOCK_HASH_LENGTH &&
+      bs58.decode(hash).length === DECODED_BLOCK_HASH_LENGTH
+    );
+  }
+
+  /** @inheritdoc */
+  isValidPrivateKey(key: string | Uint8Array): any {
+    if (typeof key === 'string') {
+      key = this.base58ToUint8Array(key);
+    }
+    try {
+      return !!Keypair.fromSecretKey(key);
+    } catch (e) {
+      return false;
+    }
+  }
+
+  /** @inheritdoc */
+  isValidPublicKey(key: string): boolean {
+    try {
+      return (
+        !!key &&
+        new RegExp('[1-9A-HJ-NP-Za-km-z]{32,44}').test(key) &&
+        PublicKey.isOnCurve(new PublicKey(key).toBuffer())
+      );
+    } catch (e) {
+      return false;
+    }
+  }
+
+  /** @inheritdoc */
+  isValidSignature(signature: string): boolean {
+    return !!signature && bs58.decode(signature).length === DECODED_SIGNATURE_LENGTH;
```

Comment:
Let's try to use base64 if we can. Not sure if web3 supports this or not, but i hope it does.

base58 is marked as deprecated because it is slower: https://docs.solana.com/developing/clients/jsonrpc-api#sendtransaction.

Reply:
Encoding/serializing the whole tx to base 64 before sending is what that doc talks about. (other coins use hex, so we call it txHex)

But the  encoding of signatures(which are used as txHash/ID too) and pubKeys are in base58